presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -316,8 +322,9 @@ private static void updateQueryOutputLocations(QueryStateMachine queryStateMachi
         Optional<int[]> bucketToPartition;
         PartitioningHandle partitioningHandle = plan.getFragment().getPartitioning();
         if (partitioningHandle.equals(SOURCE_DISTRIBUTION)) {
+            Map<PlanNodeId, SplitSource> splitSources = splitSourceFactory.createSplitSources(plan.getFragment(), session);
```

Comment:
Per discussion, this is a refactor so we still make split source created eagerly when all stage execution are created. Do we want to add a TODO or comment explain ? ditto for line 345.